README.md

```diff
@@ -13,3 +13,173 @@ PEP750 based t strings for both SSR and FE
+<div data-a="1" data-b="2" data-c="3">
+  <!-- some content -->
+</div>
+```
+
+Last, but not least, `@events` are currently specially handled as well, such as `@click`, `@pointerover` and every other standard *event* will be translated into a specialized listener that will either work or silently do nothing unless instrumented/orchestrated explicitly.
+
+```html
+<button @click={my_click_handler}>
+  click me
+</button>
+
+<!-- will result into -->
+<button onclick="self.python_listeners?.[0](event)">
+  click me
+</button>
+```
+
+Currently experimental, we've already managed to bring real Python listeners to the browser via `dill` module and [PyScript](https://pyscript.net/) ability to bootstrap [pyodide](https://pyodide.org/en/stable/) on the front end, but any custom logic able to map listeners to actual actions on the page could work similarly, if not better.
+
+
+### Content
+
+Runtime content can be placed almost anywhere and it could represent a string, number, node returned by `html` or `svg` utility or a callback that will be invoked to return any of these values or, ultimately, a `list` or a `tuple` that contains any previously mentioned value, or a component.
+
+```html
+<div>
+  Some {'text'}.
+  Some {lambda_or_function}
+  <ul>
+    {[
+      html(t'<li>{'a'}</li>'),
+      html(t'<li>b</li>'),
+      html(t'<li>c {sep} d</li>'),
+    ]}
+  </ul>
+  <{MyComponent} a='1' b={2} />
+</div>
+```
+
+
+### Components
+
+Differently from functions found as interpolation value within the content, a component is a function, or a class, that will be invoked, or instiated with 2 arguments, `props` and `children`, but it requires to be present right after an opening `<` char, otherwise it won't receive any value:
+
+```python
+def MyComponent(props, children):
+  # props['a'] == 1 and props.b == 2
+  # children == [<p />, <p />]
+  return html(t'<div data={props}>{children}</div>')
+
+print(
+  str(
+    html(t'''
+      <{MyComponent} a="1" b={2}>
+        <p>first element {'child'}</p>
+        <p c={3}>second element child</p>
+      </>
+    ''')
+  )
+)
+```
+
+The output that will result is:
+
+```html
+<div data-a="1" data-b="2">
+  <p>first element 'child'</p>
+  <p c="3">second element child</p>
+</div>
+```
+
+where all `children` will be passed along already resolved and all `props` will contain every "*attribute*" defined at the component level: *props* is just a special *dictionary* that allows both `props.x` and `props['x']` ways to read its own values.
+
+
+### A note about special syntax
+
+In these examples it is possible to note *self-closing tags*, such as `<div />` or others, but also a special *closing-tag* such as `</>` or `<//>` (these are the same).
+
+The `@` attribute for events is also not standard, but it helps explicitly distinguish between what could be an actual *JS* content for a real `onclick`, as opposite of being something "*magic*" that needs to be orchestrate @ the *Python* level.

```

Comment:
We could add a note that this would have no meaning in SSR.

Reply:
why is that? my current example works with Python on the CLI too, via `dill`, it's not a pyodide in pyodide thing, it's actually Python from the CLI exported to Python on the page ... of you meant `onclick` for the *JS* side of affairs? 'cause you can pass valid JS as string in there too ... `t'<button onclick={'alert(123)'} />'` would land a button with an `onclick` listener ... should I clarify this more or what part doesn't work SSR? 🤔